utils/githooks/find_base.sh

```diff
@@ -23,9 +23,22 @@ if command -v gh > /dev/null 2>&1; then
         fi
     fi
 else
-    # With no 'gh' command installed then check against origin/master.
+    # With no 'gh' command installed, use the "closest" branch as the target,
+    # calculated as the sum of the commits this branch is ahead and behind.
     # shellcheck disable=SC2034
-    TARGET="$ORIGIN"/master
+    all_bases=("master" "release/2.4" "feature/cat_recovery" "feature/multiprovider")
+    TARGET="$ORIGIN/master"
+    min_diff=-1
+    for base in "${all_bases[@]}"; do
+        git rev-parse --verify "$base" 2>/dev/null || continue
+        commits_ahead=$(git log --oneline "$ORIGIN/$base..HEAD" | wc -l)
+        commits_behind=$(git log --oneline "HEAD..$ORIGIN/$base" | wc -l)
+        commits_diff=$(($commits_ahead + $commits_behind))
```

Comment:
(lint) $/${} is unnecessary on arithmetic variables. [SC2004]
(lint) $/${} is unnecessary on arithmetic variables. [SC2004]